pinc/Project.inc

```diff
@@ -777,24 +787,30 @@ EOS;
             $values[] = "('$this->projectid', '$state')";
         }
         $values = join(', ', $values);
-
-        mysqli_query(DPDatabase::get_connection(), "
+        $sql = "
             INSERT INTO project_holds
             VALUES $values
-        ") or die(DPDatabase::log_error());
+        ";
+        DPDatabase::query($sql);
         log_project_event($this->projectid, $pguser, 'add_holds', join($states, ' '));
     }
 
     function remove_holds($states)
     {
         global $pguser;
 
-        $states_str = surround_and_join( $states, "'", "'", ", " );
-        mysqli_query(DPDatabase::get_connection(), "
+        $projectid_clause = sprintf(
+            "projectid='%s'",
+            DPDatabase::escape($this->projectid)
+        );
+        $states_str = surround_and_join($states, "'", "'", ", ");
```

Comment:
`surround_and_join()` doesn't escape any of the states. It would probably be easiest if we built this the same way `add_holds()` does.

Reply:
Yeah, it's better for 'symmetry' between the two implementations too. Done.